results_dashboard.py

```diff
@@ -1,20 +1,112 @@
 
 if __name__ == "__main__":
+    # Argument/config parsing
+    parser = configargparse.ArgParser(
+        auto_env_var_prefix='',
+        default_config_files=[
+            '/etc/seamm/seamm.ini',
+            '~/.seamm/seamm.ini',
+        ]
+    )
+    parser.add_argument(
+        '--seamm-configfile',
+        is_config_file=True,
+        default=None,
+        help='a configuration file to override others'
+    )
+
+    # Options for the dashboard
+    parser.add_argument(
+        "--log-level",
+        default='INFO',
+        choices=[
+            'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'
+        ],
+        type=str.upper,
+        help="the logging level for the dashboard"
+    )
+    parser.add_argument(
+        "--console-log-level",
+        default='INFO',
+        choices=[
+            'CRITICAL', 'ERROR', 'WARNING', 'INFO', 'DEBUG', 'NOTSET'
+        ],
+        type=str.upper,
+        help="the logging level for the dashboard console"
+    )
+    parser.add_argument(
+        '--mode',
+        choices=['production', 'development', 'testing'],
+        type=str.lower,
+        default='production',
+        help='the mode to run in'
+    )
+    parser.add_argument(
+        "--datastore",
+        dest="datastore",
+        default='.',
+        action="store",
+        env_var='SEAMM_DATASTORE',
+        help="The datastore (directory)."
+    )
+    parser.add_argument(
+        "--database",
+        dest="database",
+        default='seamm.db',
+        action="store",
+        env_var='SEAMM_DATABASE',
+        help="The database file."
+    )
+    parser.add_argument(
+        "--logdir",
+        dest="logdir",
+        default='%datastore%/logs',
+        action="store",
+        env_var='SEAMM_LOGDIR',
+        help="The directory for logging"
+    )
+    parser.add_argument(
+        '--initialize',
+        action='store_true',
+        help='initialize the database from scratch'
+    )
+
+    options, unknown = parser.parse_known_args()
+
+    # Setup logging
+    setup_logging(options)
+    logger = logging.getLogger()
 
-    app = create_app(config_name)
+    # And proceed. First check if the database exists
+    db_file = os.path.join(
+        os.path.expanduser(options.datastore), options.database
+    )
+    initialize = not os.path.exists(db_file)
+    
+    if options.initialize and os.path.exists(db_file):
+        os.remove(db_file)
 
-    location = app.config['SQLALCHEMY_DATABASE_URI']
-    db_path = location.replace('sqlite:///', '')
-    location = os.path.dirname(db_path)
+    app = create_app(options.mode)
 
-    with app.app_context():        
-        create_datastore(location)
+    if initialize or options.initialize:
+        logger.info('Initializing the database ({})'.format(db_file))
+        t0 = time.perf_counter()
+        with app.app_context():        
+            n_projects, n_jobs = create_datastore(
+                os.path.join(options.datastore, 'projects')
```

Comment:
This previously used `app.config['SQLALCHEMY_DATABASE_URI']` which ensured that the created database was the on the app was using.

Reply:
See note above.